File: java8/src/main/java/com/example/Point.java
package com.example;

/**
 * Created by evacchi on 28/02/15.
 */
class Point {
    double x,y;

    public Point(double x, double y) {
        this.x = x; this.y = y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public Point plus(Point p2) {
        return new Point(x + p2.getX(), y + p2.getY());
    }
    public Point minus(Point p2) {
        return new Point(x - p2.getX(), y - p2.getY());
    }
    public Point div(double d) {
        return new Point(x/d, y/d);
    }
    public Double getModulus() { return Math.sqrt(sq(x) + sq(y)); }
    private double sq(double x) { return x*x; }

    @Override
    public String toString() {
        return "(" +x+", "+y+")";
    }
}

File: kotlin/src/main/java/com/example/Point.kt
package com.example

/**
 * Created by evacchi on 26/02/15.
 */
data class Point(var x: Double, var y: Double) {

    fun div(d: Double) = Point(x/d, y/d)
    fun divAssign(d: Double) {
        x /= d
        y /= d
    }
    fun plus(p2: Point) = Point(x+p2.x, y+p2.y)
    fun plusAssign(p2: Point) {
        x += p2.x
        y += p2.y
    }
    fun minus(p2: Point) = Point(x-p2.x, y-p2.y)
    fun minusAssign(p2: Point) {
        x -= p2.x
        y -= p2.y
    }
    private fun sq(x: Double): Double = x*x

    public val modulus: Double = Math.sqrt(sq(x) + sq(y))

}


